warehouse/query-core/src/main/java/datawave/query/jexl/visitors/BashDecorator.java

```diff
@@ -0,0 +1,221 @@
+import org.apache.commons.jexl2.parser.ASTNullLiteral;
+import org.apache.commons.jexl2.parser.ASTNumberLiteral;
+import org.apache.commons.jexl2.parser.ASTOrNode;
+import org.apache.commons.jexl2.parser.ASTSizeMethod;
+import org.apache.commons.jexl2.parser.ASTStringLiteral;
+import org.apache.commons.jexl2.parser.ASTTrueNode;
+import org.apache.commons.jexl2.parser.ASTUnaryMinusNode;
+import org.apache.commons.jexl2.parser.JexlNode;
+
+import datawave.query.jexl.JexlASTHelper;
+import datawave.query.jexl.nodes.QueryPropertyMarker;
+
+/**
+ * Decorates a Jexl Query with bash coloring. To be used in viewing Jexl Queries in bash.
+ */
+public class BashDecorator implements JexlQueryDecorator {
+    
+    protected static final String AND_COLOR = "\\e[38;5;21m"; // dark blue
+    protected static final String ARITHMETIC_OP_COLOR = "\\e[38;5;10m"; // green teal
+    protected static final String ASSIGN_COLOR = "\\e[38;5;201m"; // light pink
+    protected static final String BOOLEAN_COLOR = "\\e[38;5;210m"; // orange pink
+    protected static final String EQUAL_COLOR = "\\e[38;5;51m"; // cyan
+    protected static final String ER_COLOR = "\\e[38;5;214m"; // light orange
+    protected static final String FIELD_COLOR = "\\e[38;5;11m"; // light yellow
+    protected static final String FUNCTION_COLOR = "\\e[38;5;202m"; // dark orange
+    protected static final String GREATER_EQUAL_COLOR = "\\e[38;5;129m"; // bright purple
+    protected static final String GREATER_THAN_COLOR = "\\e[38;5;69m"; // blue purple
+    protected static final String LESS_EQUAL_COLOR = "\\e[38;5;94m"; // brown orange
+    protected static final String LESS_THAN_COLOR = "\\e[38;5;2m"; // dark green
+    protected static final String MARKER_COLOR = "\\e[38;5;108m"; // green blue
+    protected static final String METHOD_COLOR = "\\e[38;5;91m"; // dark purple
+    protected static final String NAMESPACE_COLOR = "\\e[38;5;135m"; // dull purple
+    protected static final String NC = "\\e[0m"; // No color
+    protected static final String NOT_COLOR = "\\e[38;5;162m"; // dark pink
+    protected static final String NOT_EQUAL_COLOR = "\\e[38;5;88m"; // dark red
+    protected static final String NR_COLOR = "\\e[38;5;46m"; // light green
+    protected static final String NULL_COLOR = "\\e[38;5;229m"; // yellow white
+    protected static final String NUMBER_COLOR = "\\e[38;5;225m"; // pink white
+    protected static final String OR_COLOR = "\\e[38;5;1m"; // light red
+    protected static final String STRING_COLOR = "\\e[38;5;33m"; // light blue
+    
+    @Override
+    public void apply(StringBuilder sb, ASTAdditiveOperator node) {
+        sb.append(ARITHMETIC_OP_COLOR + node.image + NC);
+    }
+    
+    @Override
+    public void apply(StringBuilder sb, ASTAndNode node, Collection<String> childStrings, boolean needNewLines) {
+        sb.append(String.join(AND_COLOR + " && " + NC + (needNewLines ? NEWLINE : ""), childStrings));
```

Comment:
Probably not a big deal, but I would have only colored the '&&', not the spaces.  This doesn't matter when you're changing the font color, but it might look odd if we're coloring the background.